webapp/src/pages/boardPage/boardPage.tsx

```diff
@@ -101,23 +98,28 @@ const BoardPage = (props: Props): JSX.Element => {
             teamId: boardTeamId,
             boardId,
         }))
-
-        // and set it as most recently viewed board
-        UserSettings.setLastBoardID(boardTeamId, boardId)
-
-        if (viewId && viewId !== '0') {
-            dispatch(setCurrentView(viewId))
-            UserSettings.setLastViewId(boardId, viewId)
-        }
     }, [])
 
     useEffect(() => {
-        dispatch(loadAction(match.params.boardId))
+        if (match.params.boardId) {
+            dispatch(loadAction(match.params.boardId))
+
+            // set the active board
+            dispatch(setCurrentBoard(match.params.boardId))
 
-        if (match.params.boardId && me) {
-            loadOrJoinBoard(me.id, teamId, match.params.boardId, match.params.viewId)
+            // and set it as most recently viewed board
+            UserSettings.setLastBoardID(teamId, match.params.boardId)
+
+            if (match.params.viewId && match.params.viewId !== '0') {
+                dispatch(setCurrentView(match.params.viewId))
+                UserSettings.setLastViewId(match.params.boardId, match.params.viewId)
+            }
+
+            if (!props.readonly && me) {
+                loadOrJoinBoard(me.id, teamId, match.params.boardId)
+            }
         }
-    }, [teamId, match.params.boardId, match.params.viewId])
+    }, [teamId, match.params.boardId, match.params.viewId, me])
```

Comment:
nit: probably a super tiny optimization is to pass use the `me?.id` as dependency, that way your user changes doesn't necesarilly retrigger this effect.